project/Build.scala

```diff
@@ -131,6 +135,26 @@ object Build {
     }
     .dependsOnSource(nir)
     .dependsOnSource(util)
+    .dependsOn(testingCompilerInterface % "test")
+    .zippedSettings(Seq("testingCompiler", "scalalib")) {
+      case Seq(testingCompiler, scalalib) =>
+        Test / javaOptions ++= {
+          val nscCompilerJar =
+            (Compile / Keys.`package`).value.getAbsolutePath()
+          val testingCompilerCp =
+            (testingCompiler / Compile / fullClasspath).value.files
+              .map(_.getAbsolutePath)
+              .mkString(pathSeparator)
+          val scalalibCp = (scalalib / Compile / fullClasspath).value.files
+            .map(_.getAbsolutePath)
+            .mkString(pathSeparator)
+          Seq(
+            "-Dscalanative.nscplugin.jar=" + nscCompilerJar,
+            "-Dscalanative.testingcompiler.cp=" + testingCompilerCp,
+            "-Dscalanative.nativeruntime.cp=" + scalalibCp
+          )
+        },
+    }
```

Comment:
I think we can potentially just use natilib instead of javalib here. This way we would not need to build javalib/auxlib/clib/posixlib/windowslib which has lots of sources. 
```suggestion
    .zippedSettings(Seq("testingCompiler", "nativelib")) {
      case Seq(testingCompiler, nativelib) =>
        Test / javaOptions ++= {
          val nscCompilerJar =
            (Compile / Keys.`package`).value.getAbsolutePath()
          val testingCompilerCp =
            (testingCompiler / Compile / fullClasspath).value.files
              .map(_.getAbsolutePath)
              .mkString(pathSeparator)
          val nativelibCp = (scalalib / Compile / fullClasspath).value.files
            .map(_.getAbsolutePath)
            .mkString(pathSeparator)
          Seq(
            "-Dscalanative.nscplugin.jar=" + nscCompilerJar,
            "-Dscalanative.testingcompiler.cp=" + testingCompilerCp,
            "-Dscalanative.nativeruntime.cp=" + nativelibCp
          )
        },
    }
```

Reply:
Indeed it works 👍